distribution/sdk/sample-cometd/cometd-java-example/src/main/java/org/codice/ddf/sdk/cometd/AsyncClient.java

```diff
@@ -0,0 +1,231 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package org.codice.ddf.sdk.cometd;
+
+import java.net.MalformedURLException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.cometd.bayeux.Message;
+import org.cometd.bayeux.client.ClientSessionChannel;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.ClientTransport;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AsyncClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AsyncClient.class);
+
+    private static final String ACTIVITIES_CHANNEL = "/ddf/activities";
+    private static final String ALL_ACTIVITIES = ACTIVITIES_CHANNEL + "/**";
+    private static final String NOTIFICATIONS_CHANNEL = "/ddf/notifications";
+    private static final String ALL_NOTIFICATIONS = NOTIFICATIONS_CHANNEL + "/**";
+    private static final String DOWNLOADS_CHANNEL = NOTIFICATIONS_CHANNEL + "/downloads";
+    private static final String QUERY_SERVICE = "/service/query";
+    private static final String COMETD_CONTEXT = "/cometd";
+    private static final String DOWNLOAD_CONTEXT = "/services/catalog/sources/";
+    private static final String DOWNLOAD_TRANSFORM = "?transform=resource";
+
+    private static String asyncClientId;
+    private static Map<String, Object> queryResponse;
+    private static Map<String, Object> activities;
+    private static Map<String, Object> notifications;
+
+    private String url;
+    private BayeuxClient client;
+    private Map<String, Object> emptyMessage = new HashMap<>();
+
+    /**
+     * Creates a ddf async client with default channel subscriptions
+     * @param url - target ddf url
+     * @param disableValidation - disable ssl hostname validation
+     * @throws Exception
+     */
+    public AsyncClient(String url, Boolean disableValidation) throws Exception {
+
+        this.url = url;
+
+        HttpClient httpClient = new HttpClient();
+
+        httpClient.start();
+
+        Map<String, Object> options = new HashMap<String, Object>();
+        ClientTransport transport = new LongPollingTransport(options, httpClient);
+
+        // DisableValidation
+        if (disableValidation) {
+            doTrustAllCertificates();
+        }
+
+        this.client = new BayeuxClient(url + COMETD_CONTEXT, transport);
+
+        client.handshake(new ClientSessionChannel.MessageListener() {
```

Comment:
May want to consider doing a static import of `ClientSessionChannel.MessageListener` since we're using it in many places.